### PR TITLE
Refactor javaharness code

### PR DESCRIPTION
**Pull request by yuangu-google**
- Move clipboard service and toast service out of api/ into demo/services
- Move builtinparticles to demo/particles
- Split code under webimpl into web/impl and web/demo
- Split code under impl into android/impl and web/impl

### CODE REVIEW

File: src/javaharness/java/arcs/api/BUILD [hunk 1]
```diff
@@ -9,15 +9,15 @@ load("@com_google_j2cl//build_defs:rules.bzl", "j2cl_library")
 load("@build_bazel_rules_android//android:rules.bzl", "android_library")
 
 android_library(
-    name = "api",
+    name = "api-android",
```
Comment (shaper) [inline src/javaharness/java/arcs/api/BUILD]: Why the '-android' suffixes here (and below)? Is this commonly done, I haven't seen before.

Comment (yuangu-google) [reply src/javaharness/java/arcs/api/BUILD]: I expect only general Java library without suffix. Android versions should end with -android.

File: package-lock.json [hunk 2]
```diff
@@ -2516,7 +2516,7 @@
     },
     "browserify-aes": {
       "version": "1.2.0",
-      "resolved": "http://registry.npmjs.org/browserify-aes/-/browserify-aes-1.2.0.tgz",
+      "resolved": "https://registry.npmjs.org/browserify-aes/-/browserify-aes-1.2.0.tgz",
```
[the remark this answers is not in the document]
Comment (yuangu-google) [reply package-lock.json]: Reverted.

File: src/javaharness/java/arcs/api/BUILD (hunk 1 above)
Comment (cromwellian) [reply src/javaharness/java/arcs/api/BUILD]: This is fine because it aligns with the -j2cl target being web.

**Pull request merged**